spec/integration/command_macros_spec.rb

```diff
@@ -46,4 +47,34 @@
     user = repo.users.by_id(user.id).one
     expect(user.name).to eql('Jane')
   end
+
+  it 'allows to pass a block to configure command class' do
+    repo = Class.new(ROM::Repository[:users]) do
+      command(:create) do
+        use :upcase_name
+      end
+    end.new(rom)
+
+    user = repo.create(name: 'Jane')
+
+    persisted_user = repo.users.by_id(user.id).one
+    expect(persisted_user.name).to eql('JANE')
+  end
+
+  it 'allows to pass a block with view syntax' do
+    repo = Class.new(ROM::Repository[:users]) do
+      command :create
+      command update: :by_id do
+        use :upcase_name
+      end
```

Comment:
So you basically need this for dumping some values `to_json` if I read this correctly. I believe this can be handled by the upcoming schema with coercible types. Ie I already added `SQL::Types::PG::JSON` for pg-array/hash. We can easily add more types like that.